src/Components/Forms/CreateActivationKeyForm.js

```diff
@@ -0,0 +1,176 @@
+      submitForm(name, role, serviceLevel, usage);
+    } else {
+      setValidated('error');
+    }
+  };
+
+  const validateName = (value) => {
+    if (value.length === 0) {
+      setValidated('error');
+    } else if (!value.match(nameRegex)) {
+      setValidated('error');
+    } else {
+      setValidated('success');
+      setName(value);
+    }
+  };
+
+  if (isSuccess) {
+    dispatch(
+      addNotification({
+        variant: 'success',
+        title: ' Activation Key was created successfully',
+        description: '',
+        dismissable: true,
+      })
+    );
+    handleModalToggle();
+  } else if (isError) {
+    dispatch(
+      addNotification({
+        variant: 'danger',
+        title: 'An Error Occurred',
+        description: 'Activation Key was not created, please try again',
+        dismissable: true,
+        autoDismiss: false,
+      })
+    );
+    handleModalToggle();
+  }
+
+  return (
+    <Form id="create-activation-key-form" onSubmit={handleSubmit}>
+      <FormGroup
+        label="Name"
+        isRequired
+        helperText="Enter the name of the activation key. Must start with a letter and end with a letter or number. Valid characters include lowercase letters from a to z, numbers from 0 to 9, and hypnes ( - )."
+      >
+        <TextInput
+          id="activation-key-name"
+          label="Name"
+          isRequired
+          type="text"
+          validated={validated}
+          onChange={validateName}
+        />
+      </FormGroup>
+      {!isLoading && !error && (
+        <ActivationKeysFormSelect
+          data={data.roles}
+          onSelect={setRole}
+          label="Role"
+          popover={
+            <Popover
+              bodyContent={
+                <TextContent>
+                  <Text component={TextVariants.p}>
+                    Role is used to categorize systems by the workload on the
+                    system
+                  </Text>
+                  <Text component={TextVariants.p}>
+                    Subscription Watch can help you filter and report by these
+                    items.
+                  </Text>
+                  <Text component={TextVariants.p}>
+                    Only roles available to your account are shown.
+                  </Text>
+                </TextContent>
+              }
+            >
+              <HelpIcon />
+            </Popover>
+          }
+          helperText="Select the required role from the list. The list only contains roles available to the activation key."
+        />
+      )}
+      {!isLoading && !error && (
+        <ActivationKeysFormSelect
+          data={data.serviceLevel}
+          onSelect={setServiceLevel}
+          label="Service Level Agreement (SLA)"
+          popover={
+            <Popover
+              bodyContent={
+                <TextContent>
+                  <Text component={TextVariants.p}>
+                    Service Level Agreement determines the level of support for
```

Comment:
missing the (SLA) in parentheses
"Service Level Agreement (SLA) determines ..."